### ISSUE TITLE
Failed "Almost Equal" assertion breaks UTF-8 compatibility

### ISSUE THREAD

**Issue by plundeen** [JKISoftware/Caraya#131]
We leverage Caraya for the unit test stage of our Gitlab CI / CD pipelines. We just noticed an issue where we had unit test failures in our pipeline, but the Gitlab Merge Request UI was not rendering the JUnit XML test report. After some digging, we found the error was
`JUnit XML parsing failed: 49:41: FATAL: Input is not proper UTF-8, indicate encoding ! Bytes: 0xB1 0x20 0x30 0x2E`

It looks like the `xB1` character is coming from an `Assert Almost Equal` failure (the "±" character):

```
<failure message="{Expected: 271.525516 ± 0.000001, Measured: 0.000000}">"FAIL"</failure>
```
Viewing the XML in Notepad++ also renders that character as `xB1`.

Can you search for any other special characters that might violate UTF-8 formatting and replace them to ensure report parsing tools work correctly?

**Comment by francois-normandin** [comment]
This will be fixed in release 1.2.
Thanks for the report @plundeen .
A quick look in the code didn't allow me to find more instances, but Caraya still does not support a user-defined non-ascii message, unless we make the UI and Test Report classes UTF compatible.

As a side note... In case you need full UTF support in the reports, as stated above, you can override the base report class and inject your UTF-supported code in the Test Suite. Of course, this kind of functionality would be a great addition to Caraya for all to benefit. I'll open a new issue to track the feature request of supporting UTF8 natively for user-defined inputs.

**Comment by francois-normandin** [comment]
Included in 1.2.0 to be released soon
https://github.com/JKISoftware/Caraya/milestone/6
